inst/odin/model.R

```diff
@@ -79,23 +79,46 @@ seedrate0 <- user(0.75)
 dseedrate0 <- user(0)
 seedrate_sd <- user(0.75) #2.99 #sd of random walk of daily diff in seedrate
 #~ vacc_freq <- user(1)
-vacc_start_day <- user(91)
-vacc_start_day2 <- user(91+45) #TODO find date 2nd dose provided
+
 vacc_targetted <- user(.8) # prop vacc targetted vs random
 vacc_efficacy <- user(0.78) # Bertran et el, one dose
 vacc_efficacy2 <- user(1.00) # 2 doses, TODO values
-vacc_doses <- user(50e3) # Assume 50k doses in UK
-vacc_doses2 <- user(15e3) # TODO find number of 2nd doses
 cumulative_partners_days <- user(90)
-vacc_duration <- user(55) ## 2022-08-30 - 2022-07-06
-vacc_duration2 <- user(55) ##  find dates 2nd dose provided
+
+# time-varying doses
+vacc_start_day_step[] <- user()
+vacc_start_day2_step[] <- user()
+vacc_doses_step[] <- user()
+vacc_doses2_step[] <- user() 
+vacc_duration_step[] <- user()
+vacc_duration2_step[] <- user() 
+
+dim(vacc_start_day_step) <- user()
+dim(vacc_start_day2_step) <- user()
+dim(vacc_doses_step) <- user()
+dim(vacc_doses2_step) <- user() 
+dim(vacc_duration_step) <- user()
+dim(vacc_duration2_step) <- user() 
 
 exp_noise <- user(1e6) # ignore.unused
 kappa_cases <- user(1) # ignore.unused
 rho_travel <- user(0.5) # ignore.unused
 use_new_compare <- user(0) # ignore.unused
 
+
+
 # new vacc if within schedule (after delay, taking effect)
+vacc_period_next <-  vacc_period + if (as.integer(time) == vacc_fin_day) 1 else 0
```

Comment:
Using `as.integer(time)==vacc_fin_day` may cause problems because the time step is now now customisable and usually less than 1. 
This condition would be triggered multiple times for a given `vacc_fin_day`, e.g. 
at `vacc_fin_day, vacc_fin_day+dt,vacc_fin_day+2*dt` etc

Reply:
We can change the condition to `time > vacc_fin_day & (time-dt)<vacc_fin_day`